clib/src/main/scala/scala/scalanative/libc/bsd/kevent.scala

```diff
@@ -0,0 +1,104 @@
+package scala.scalanative.libc
+package bsd
+
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+
+@extern
+@define("__SCALANATIVE_C_KEVENT")
```

Comment:
Same issues as in the #4624 - let's use it to conditionally compile our glue code only if it's actually needed. 
Let's move all to posixlib `scala.scalanative.bsd`, at that point this definition can be renamed accordingly. 
Beside that it looks good, it would be also useful for other usages in the future